app/services/auth.js

```diff
@@ -12,6 +12,8 @@ import URLPolyfill from 'travis/utils/url';
 
 const proVersion = config.featureFlags['pro-version'];
 
+let includes = [];
```

Comment:
I’m wondering why this is out here rather than in the `refreshUserData` function; probably a good reason, maybe worth explaining?

Reply:
This is for consistency reasons. Kind of a "singleton"-ish behaviour - the array is shared between all instances and `refreshUserData` calls. Each call to `refreshUserData` that has new `includes`, appends missing ones to this array and never deletes. This is to allow requested data survive future refreshes. Let's say, on the Help page the app requests `user` with `user.emails` included and shows them on UI. Something in background (let's say, pusher event or something else) calls `refreshUserData` via `auth` service without `user.emails`, which will override the data, remove `emails` and break the Help page UI. The current implementation aims to avoid such cases. 
Does it makes sense in your opinion? Or would you prefer single-request enhancements?

Reply:
oh, making sure it doesn’t override is a great reason I didn’t consider. Maybe some commentary would help communicate this to future visitors to this code, like:

```
// Ensures future user-fetching doesn’t erase emails attribute
let persistentIncludes = [];
```

or something?

Reply:
Absolutely! Thanks, I'll add a comment 👍